pkg/agent/openflow/pipeline.go

```diff
@@ -2664,8 +2738,12 @@ func (f *featurePodConnectivity) l3FwdFlowToLocalPodCIDR() []binding.Flow {
 	var flows []binding.Flow
 	for ipProtocol, cidr := range f.localCIDRs {
 		// This generates the flow to match the packets destined for local Pods without RewriteMACRegMark.
-		flows = append(flows, L3ForwardingTable.ofTable.BuildFlow(priorityLow).
-			Cookie(cookieID).
+		fb := L3ForwardingTable.ofTable.BuildFlow(priorityLow).
+			Cookie(cookieID)
+		if f.connectUplinkToBridge {
+			fb = fb.MatchRegFieldWithValue(VLANIDField, 0)
```

Comment:
Match VLAN field instead of register directly?

Reply:
As replied https://github.com/antrea-io/antrea/pull/3247#discussion_r829638323 , VLAN field is not available if in_port is access port.